packages/contracts/contracts/testing/ONFT721Reward.sol

```diff
@@ -0,0 +1,111 @@
+
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@snickerdoodlelabs/erc7529/contract/ERC7529.sol";
+import "@layerzerolabs/solidity-examples/contracts/token/onft721/ONFT721.sol";
+
+contract ONFT721Reward is ERC721Burnable, AccessControl, ONFT721, ERC7529 {
+    using Counters for Counters.Counter;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    Counters.Counter private _tokenIdCounter;
+    string public baseURI;
+
+    constructor(string memory name, string memory symbol, string memory baseUri, uint256 minGasToTransfer, address lzEndpoint) ONFT721(name, symbol, minGasToTransfer, lzEndpoint) {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _grantRole(MINTER_ROLE, msg.sender);
+        setBaseURI(baseUri);
+    }
+
+    function safeMint(address to) public onlyRole(MINTER_ROLE) {
+        uint256 tokenId = _tokenIdCounter.current();
+        _tokenIdCounter.increment();
```

Comment:
Just remembered that this needs to be updated to be a random token id rather than counting up. 
This is so that when we do cross chain transfer, the counter doesn't get messed up on the receiving contract. 

Eg. we minted 3 tokens on contract A, token ids 0, 1 and 2 exist. We cross chain transfer 2, burn it on contract A, and mint token id on contract B. Now when contract B does mints, it would start from 0, mints 1 and when it hits token id 2, it would already exist on contract B.